lib/trino-hdfs/src/main/java/io/trino/hdfs/HdfsEnvironment.java

```diff
@@ -68,6 +70,15 @@ public HdfsEnvironment(
         this.newDirectoryPermissions = config.getNewDirectoryFsPermissions();
     }
 
+    @PreDestroy
+    public void shutdown()
+            throws IOException
+    {
+        FileSystemFinalizerService.getInstance().shutdown();
+        stopFileSystemStatsThread();
+        TrinoFileSystemCache.INSTANCE.closeAll();
```

Comment:
Shouldn't this be called only when classloader is isolated? e.g. not during integration tests?

Reply:
How do we run multiple tests in parallel today without the file systems interfering with each other?